### PR TITLE
Updating getSlotCounts to use sessions allowed

### PR DESCRIPTION
**Pull request by mitchloudenbeck**
- [X] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

Switching getSlotCounts to use the maxSessions when counting the total
slots. Old method would count all browser options so the free/total counts
could be much higher than what was actually available.

### CODE REVIEW

File: java/server/src/org/openqa/grid/web/servlet/HubStatusServlet.java [hunk 1]
```diff
@@ -136,16 +136,14 @@ private JsonObject getResponse(HttpServletRequest request) throws IOException {
...
 
     for (RemoteProxy proxy : getRegistry().getAllProxies()) {
-      for (TestSlot slot : proxy.getTestSlots()) {
-        if (slot.getSession() == null) {
-          freeSlots += 1;
-        }
-
-        totalSlots += 1;
-      }
+      totalSlots += proxy.getTestSlots().size() > proxy.getMaxNumberOfConcurrentTestSessions() ?
```
Comment (lukeis) [inline java/server/src/org/openqa/grid/web/servlet/HubStatusServlet.java]: totalSlots += Math.min(proxy.getMaxNumberOfConcurrentTestSessions(), proxy.getTestSlots().size());

Comment (mach6) [inline java/server/src/org/openqa/grid/web/servlet/HubStatusServlet.java]: don't need `+=` here. `=` will do

File: java/server/src/org/openqa/grid/web/servlet/HubStatusServlet.java [hunk 2]
```diff
@@ -136,16 +136,14 @@ private JsonObject getResponse(HttpServletRequest request) throws IOException {
...
+        proxy.getMaxNumberOfConcurrentTestSessions() : proxy.getTestSlots().size();
+      usedSlots += proxy.getTotalUsed();
```
Comment (mach6) [inline java/server/src/org/openqa/grid/web/servlet/HubStatusServlet.java]: same comment about `+=`

**Comment by mach6** [comment]
LGTM, once comments addressed. :)

File: java/server/src/org/openqa/grid/web/servlet/HubStatusServlet.java (hunk 1 above)
Comment (mitchloudenbeck) [inline java/server/src/org/openqa/grid/web/servlet/HubStatusServlet.java]: Don't we still want += here? It is summing all the slots/sessions of all the nodes connected to the hub.

Comment (lukeis) [inline java/server/src/org/openqa/grid/web/servlet/HubStatusServlet.java]: reducing this further... could just get rid of the two other variables and just have:

    freeSlots += Math.min(proxy.getMaxNumberOfConcurrentTestSessions(), proxy.getTestSlots().size()) - proxy.getTotalUsed();

and yeah, i agree still need the `+=` :)

**Pull request merged**